### PR TITLE
Add source of constituent data to projects and samples

### PR DESCRIPTION
**Pull request by katieemelianova**
Add data source to samples (run elements included) and projects (samples included)

Add tests

### CODE REVIEW

File: analysis_driver/dataset.py [hunk 1]
```diff
@@ -657,6 +669,10 @@ def initialise_entity(self):
             'dataset_type': self.dataset.type,
             'dataset_name': self.dataset.name
         }
+
+        if self.dataset.type in ['project', 'sample']:
+            entity.update({'data_source': self.dataset.data_source})
```
Comment (tcezard) [inline analysis_driver/dataset.py]: This is going to raise NotImplementedError in RunDataset's case

**Review by tcezard — CHANGES_REQUESTED** [review]
`data_source` needs to be implemented in RunDataset

File: analysis_driver/dataset.py (hunk 1 above)
Comment (katieemelianova) [reply analysis_driver/dataset.py]: This I think would not raise NotImplementedError in RunDataset due to the if statement - unless I'm missing something?
I have added data_source to RunDataset but returning None instead to not have to check for dataset type

**Review by mwhamgenomics — APPROVED** [review]

**Review by tcezard — APPROVED** [review]

File: analysis_driver/dataset.py (hunk 1 above)
Comment (tcezard) [reply analysis_driver/dataset.py]: that is indeed correct

**Pull request merged**